### PR TITLE
MAINTAINERS: initial version of MAINTAINERS file

### PR DESCRIPTION
**Pull request by ioannisg**
Pushing the initial version of Zephyr MAINTAINERS file.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Data base for holding maintainer-ship info in Zephyr

- Aligns with TSC terminology (Maintainers, Collaborators)
- Covers most of the Zephyr source Tree (initial source: https://docs.google.com/spreadsheets/d/1QxfZuM1qQIXy_EJldQeS6nErt7DjHw3SZ2vfiKVdo5M/edit#gid=0)
- Adds orphaned areas as well

- ~~Requires TSC approval~~ (motion approved)

### CODE REVIEW

File: MAINTAINERS.yml
Comment (stephanosio) [inline MAINTAINERS.yml:424]: missing `status:`

File: MAINTAINERS.yml
Comment (stephanosio) [inline MAINTAINERS.yml:322]: missing `status:`

**Review by stephanosio — COMMENTED** [review]
I suppose missing `status:` is not intentional?

File: MAINTAINERS.yml
Comment (alexanderwachter) [inline MAINTAINERS.yml:564]: I think this should be @Mani-Sadhasivam

File: MAINTAINERS.yml
Comment (Mani-Sadhasivam) [inline MAINTAINERS.yml:542]: I still maintain this.

File: MAINTAINERS.yml
Comment (Mani-Sadhasivam) [inline MAINTAINERS.yml:564]: This one too

File: MAINTAINERS.yml
Comment (erwango) [inline MAINTAINERS.yml:256]: Is @ulfalizer still active?

File: MAINTAINERS.yml
Comment (erwango) [inline MAINTAINERS.yml:387]: Miss DAC

**Review by erwango — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (nashif) [inline MAINTAINERS.yml:36]: I would like to see the syntax unified here and using the same globbing we already use elsewhere in the project, having two section for files with regex support will only make it difficult to integrate with existing CI actions and workflows.

**Review by nashif — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:380]: Added DAC, checked with @martinjaeger for availability

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml]: Not sure, so I removed the entry.

**Comment by ioannisg** [comment]
The Maintainer's database should be ready for final review:

- I have updated the database based on the outcome of the Virtual F2F, see here: https://docs.google.com/document/d/1BDLlEh5_YuSiaQURvSO7JkHoF50cy9AD1kMOeDhO1ms/edit#
- I reached out to new candidate maintainers for availability, but added everyone also in review-here so they can +1 and confirm acceptance of the new role
- TSC only needs to approve _changes_ to maintenance status, marked in the database with 
`    # Provisional entry, subject to approval`
TSC only needs to approve maintainers (not added collaborators)

**Comment by ioannisg** [comment]
FYI: @ruuddw, @trond-snekvik @jhedberg @cvinayak @tejlmand @andrewboie @nashif @nvlsianpu @galak @vanwinkeljan @carlescufi @anangl @Vudentz @alexanderwachter @nordic-krch @ceolin @martinjaeger @henrikbrixandersen @tbursztyka @mnkp @microbuilder @pabigot @Mani-Sadhasivam @mbolivar-nordic @mike-scott @dcpleung @jukkar @MaureenHelm @loicpoulain @de-nordic @aescolar @pfalcon @jfischer-phytec-iot @wentongwu @nategraff-sifive @jakub-uC @erwango

**Review by trond-snekvik — APPROVED** [review]
Approving entry for Bluetooth Mesh

File: MAINTAINERS.yml
Comment (pfalcon) [inline MAINTAINERS.yml:349]: This is not correct entry. There should be 2 entries: "Subsys: Console", with include/console/ and subsys/console/, and myself as a maintainer. And "Drivers: Console" with include/drivers/console/ and drivers/console/. That's part which have wonders like handling line editing in ISR. Glad to hear it's orphaned, should be gone altogether. Feel free to keep me in "collaborators", so somebody watched what happens there.

File: MAINTAINERS.yml
Comment (alexanderwachter) [inline MAINTAINERS.yml:1030]: Don‘t we create lots of conflicts with this entry?

**Review by alexanderwachter — APPROVED** [review]

File: MAINTAINERS.yml
Comment (tbursztyka) [inline MAINTAINERS.yml:620]: jukkar could be set as maintainers there

File: MAINTAINERS.yml
Comment (tbursztyka) [inline MAINTAINERS.yml:882]: switch maintainers/collaborators: jhe is the maintainer of this part.

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:620]: Agreed, I already tried to reach him, to confirm he is available, but seems he is off-work. Not sure I want us to appoint him without him accepting. Can always be done at a later point.

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:882]: That's ok.
@jhedberg ^^

Comment (jhedberg) [reply MAINTAINERS.yml:882]: That's fine with me. I've always viewed it as us having shared maintainership, but if you want to put just one name I don't object to mine being there.

**Review by henrikbrixandersen — APPROVED** [review]
Approved my EEPROM and CAN entries.

File: MAINTAINERS.yml
Comment (avisconti) [inline MAINTAINERS.yml:652]: collaborators:
        - avisconti

File: MAINTAINERS.yml
Comment (erwango) [reply MAINTAINERS.yml:1030]: I'm happy to remove if this is not in line with general agreement and we expect there's only one assignee per PR.

File: MAINTAINERS.yml
Comment (avisconti) [reply MAINTAINERS.yml:652]: I discussed some time ago with @MaureenHelm 
I can be a collaborator here.

**Review by avisconti — CHANGES_REQUESTED** [review]

**Review by erwango — COMMENTED** [review]
- Should we keep "labels" keyword ? (since it is now covered by GH action)
- s/maintainers/maintainer
- Don't we miss a dts area?

**Comment by ioannisg** [comment]
> * Should we keep "labels" keyword ? (since it is now covered by GH action)

Good point, I'd like @nashif to comment here.

> * s/maintainers/maintainer

This was discussed in the F2F. Although we agreed that it's probably good to have a single maintainer, we might want to have multiple ones for one or several areas, so i think it is better to keep "maintainers" at the moment for backwards-compatibility. We won't have to change the grammar later, if we decide differently. Also matches with "collaborators" which is also in plural.

> * Don't we miss a dts area?
We do have a "Device Tree" area, not sure what you refer to.

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:652]: sure, i'll update this already.

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:349]: Sure @pfalcon i'll add this entries as you propose (will need to go via TSC approval)

**Review by Mani-Sadhasivam — APPROVED** [review]
LED and LoRa part LGTM. I'd like @andysan to be a collaborator if he is interested but that can be done later also.

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:882]: sure, done now

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:661]: done now, @avisconti

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:1041]: Yeap, I don't see a problem, I think @alexanderwachter might be referring to conflict for assignee-picking not conflicts in reviews.
when it comes to file/directory coverage, we do have conflicts allover the place, we just need to use some smart ways to pick assignees in PRs.

In this particular example CAN area and STM32 area both have maintainers appointed; for a PR changing an STM32 CAN drivers any of the respective maintainers may be assignees. We should be using some common sense here.
Otherwise, if a vendor area does not have an explicit maintainer, then Driver-Area maintainers become assignees for such PRs.

**Comment by pabigot** [comment]
I'm willing to be maintainer for I2C.

File: MAINTAINERS.yml
Comment (alexanderwachter) [reply MAINTAINERS.yml:1042]: It is just about the script. It would be annoying if the assignee picking is not working correctly. If we could have the maintainer as assignee and @erwango as reviewer, that would be perfect. But before such mechanism is working, we should be careful.

**Review by dcpleung — APPROVED** [review]
Approved for UART drivers, PCIe drivers, and Xtensa arch.

**Review by ceolin — APPROVED** [review]

**Comment by andysan** [comment]
> LED and LoRa part LGTM. I'd like @andysan to be a collaborator if he is interested but that can be done later also.

Happy to help out with LoRa and LoRaWAN if needed.

**Review by martinjaeger — APPROVED** [review]
DAC driver entry approved from my side.

**Review by MaureenHelm — APPROVED** [review]
Thanks @ioannisg !

File: MAINTAINERS.yml
Comment (mike-scott) [inline MAINTAINERS.yml:913]: @ioannisg LwM2M subsystem now maintained by @rlubos

**Review by mike-scott — DISMISSED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Comment by nashif** [comment]
> Good point, I'd like @nashif to comment here.

ideally we should use just this file for all this information and drop the file dedicated to labels.

**Review by wentongwu — APPROVED** [review]

**Review by anangl — APPROVED** [review]
Approved the ADC, I2S, and PWM entries.

**Comment by de-nordic** [comment]
+1 On mcumgr and VFS entries. It might be worth asking @nvlsianpu  and @pabigot  to be collaborators on VFS.

File: MAINTAINERS.yml
Comment (avisconti) [reply MAINTAINERS.yml:662]: @ioannisg 
Thanks

**Review by avisconti — APPROVED** [review]

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:914]: Great, thanks, I updated the entry. Will need, formally, TSC approval.

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:1043]: Yeap, we should take care of this, thanks @alexanderwachter

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:630]: Updated this now @jukkar @tbursztyka

**Comment by ioannisg** [comment]
> I'm willing to be maintainer for I2C.

Thanks @pabigot , updated the entry, pls, +1.

**Comment by ioannisg** [comment]
> > LED and LoRa part LGTM. I'd like @andysan to be a collaborator if he is interested but that can be done later also.
> 
> Happy to help out with LoRa and LoRaWAN if needed.

Collabs can be added at later stage

File: MAINTAINERS.yml
Comment (erwango) [inline MAINTAINERS.yml:645]: Should it be rather
```suggestion
    maintainers: 
        - anangl
```
?

**Review by erwango — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml]: Yes, thanks. Fixed now. Also similar occurrences of this problem.

File: MAINTAINERS.yml
Comment (jukkar) [inline MAINTAINERS.yml:893]: Could I be added as a collaborator here?

File: MAINTAINERS.yml
Comment (jukkar) [inline MAINTAINERS.yml:905]: and also here

File: MAINTAINERS.yml
Comment (jukkar) [inline MAINTAINERS.yml:914]: This needs to change. @rlubos could you take maintainership of the CoAP stuff?
I could be added as a collaborator here too.

File: MAINTAINERS.yml
Comment (jukkar) [inline MAINTAINERS.yml:925]: Could I be added as a collaborator here.

File: MAINTAINERS.yml
Comment (rlubos) [reply MAINTAINERS.yml:914]: @jukkar I think I can take it as well since it's tightly related to LwM2M anyway.
This might need TSC approval, right @ioannisg?

**Review by pabigot — APPROVED** [review]
accept maintainer for I2C and littlefs

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:914]: Yeap, this is right. Let me change this then.

**Comment by ioannisg** [comment]
@jukkar it's not needed to add now collaborators, can be done after we merge the initial version, without need for TSC approval

File: MAINTAINERS.yml
Comment (wentongwu) [inline MAINTAINERS.yml:981]: I think we need add @pabigot here.

Comment (ioannisg) [reply MAINTAINERS.yml:981]: Yeap, we could do this later, though.

**Comment by ioannisg** [comment]
> > Good point, I'd like @nashif to comment here.
> 
> ideally we should use just this file for all this information and drop the file dedicated to labels.

Well, this should be the case, at least for the "Area: xxx" labels, which map to code maintenance areas. 
For other types of labels, @nashif , I am not sure if we want to use this file.

**Review by carlescufi — APPROVED** [review]

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:36]: @nashif not sure I get it 100%: 
So you propose to have a single files-regex: property, so we would only be including files coming from regular expressions?

**Comment by nashif** [comment]
> Well, this should be the case, at least for the "Area: xxx" labels, which map to code maintenance areas.
> For other types of labels, @nashif , I am not sure if we want to use this file.

why do we want to do labels in 2 different places and have paths done twice? This will just lead to conflicting data. But regardless of that, I still think we should use the same format for paths that is not spread between different keys the same way we do it in the auto-labeler to avoid having to deal with 2 different formats that needs to be supported in CI.

The format suggested can be seen here: https://github.com/actions/labeler#usage

**Comment by ioannisg** [comment]
> why do we want to do labels in 2 different places and have paths done twice? This will just lead to conflicting data.

I don't know how much of a problem this is. I am only saying that it makes sense for the Maintainers' database to do labels for "areas", but we might want also to be adding different kind of labels (i.e. not related to Areas) and _that_ set of labels should probably not be listed in the MAINTAINERS.yml

I don't think this could we conflicting: we should just do "union" of whatever is generated by each script.

**Comment by nashif** [comment]
> I don't think this could we conflicting: we should just do "union" of whatever is generated by each script.

the issue now is about the files format, lets put labeling aside. We agreed to merge this as is and do the change to file format later, so I am fine with this.

**Review by nashif — APPROVED** [review]

File: MAINTAINERS.yml
Comment (abrodkin) [inline MAINTAINERS.yml:120]: Again, @vonhust is leaving so I would suggest to add me @abrodkin,  Watson (@IRISZZW) & Eugeniy (@Palmyr3) here in collaborators.

File: CODEOWNERS
Comment (abrodkin) [inline CODEOWNERS:21]: @vonhust is leaving Synopsys, so pls mention @abrodkin instead.

**Review by abrodkin — CHANGES_REQUESTED** [review]
@vonhust is leaving Synopsys so it's good to specify other people instead of him for ARC stuff.

File: MAINTAINERS.yml
Comment (ioannisg) [reply MAINTAINERS.yml:120]: Sure, I 'll make this change, since this is only about collaborators.

Comment (ioannisg) [reply MAINTAINERS.yml:120]: @evgeniy-paltsev, I suppose.

Comment (abrodkin) [reply MAINTAINERS.yml:120]: > 
> 
> @evgeniy-paltsev, I suppose.

Right, he just switched his GitHub account to be easier matched with real name :)

Comment (ioannisg) [reply MAINTAINERS.yml]: done now @abrodkin

File: CODEOWNERS
Comment (carlescufi) [reply CODEOWNERS]: This is done now @abrodkin

**Review by abrodkin — APPROVED** [review]
LGTM now, thanks

**Review by jakub-uC — APPROVED** [review]

File: MAINTAINERS.yml
Comment (bwasim) [inline MAINTAINERS.yml:619]: Typo, Should be "pcie".. Same below..

Comment (stephanosio) [reply MAINTAINERS.yml:619]: This is not a typo.

**Pull request merged**
